uk_bin_collection/uk_bin_collection/councils/MidlothianCouncil.py

```
@@ -1,178 +1,158 @@
-                soup.find("article", class_="container")
-                .find("h2", text="Search results")
-                .find_next("ul", class_="item-list item-list__rich")
+            if os.environ.get("DISPLAY") and web_driver is None:
+                try:
+                    import undetected_chromedriver as uc
+                    uc_opts = uc.ChromeOptions()
+                    uc_opts.add_argument("--no-sandbox")
+                    uc_opts.add_argument("--disable-dev-shm-usage")
+                    uc_opts.add_argument("--window-size=1920,1080")
+                    driver = uc.Chrome(options=uc_opts, version_main=146)
+                except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)